services/src/pro/datasets/storage.rs

```diff
@@ -1,31 +1,71 @@
-use crate::pro::users::UserId;
-use geoengine_datatypes::dataset::{DatasetProviderId, InternalDatasetId};
+use std::str::FromStr;
+
+use crate::error::Result;
+use crate::pro::users::{UserId, UserSession};
+use async_trait::async_trait;
+use geoengine_datatypes::{
+    dataset::{DatasetId, DatasetProviderId},
+    identifier,
+};
+#[cfg(feature = "postgres")]
+use postgres_types::{FromSql, ToSql};
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
-pub enum DatasetPermission {
-    Read,
-    Write,
-    Owner,
+identifier!(RoleId);
+
+impl From<UserId> for RoleId {
+    fn from(user_id: UserId) -> Self {
+        RoleId(user_id.0)
+    }
 }
 
 #[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
-pub struct UserDatasetPermission {
-    pub user: UserId,
-    pub dataset: InternalDatasetId,
-    pub permission: DatasetPermission,
+pub struct Role {
+    pub id: RoleId,
+    pub name: String,
+}
+
+impl Role {
+    pub fn system_role_id() -> RoleId {
+        RoleId::from_str("d5328854-6190-4af9-ad69-4e74b0961ac9").expect("valid")
+    }
+
+    pub fn user_role_id() -> RoleId {
+        RoleId::from_str("4e8081b6-8aa6-4275-af0c-2fa2da557d28").expect("valid")
+    }
```

Comment:
What is this? User's should have their own id, right?

Reply:
This is the equivalent to the user group in VAT. A user has their own id, but all the registered users get assigned this user_role additionaly for them to see all the datasets that shall be visible to all users.